src/databricks/labs/ucx/source_code/notebook.py

```diff
@@ -226,60 +231,80 @@ def read_cell_language(self, lines: list[str]) -> CellLanguage | None:
             return None
         return None
 
-    def new_cell(self, source: str) -> Cell:
-        return self._new_cell(source)
+    def new_cell(self, source: str, original_offset: int) -> Cell:
+        return self._new_cell(source, original_offset)
 
     def extract_cells(self, source: str) -> list[Cell] | None:
         lines = source.split('\n')
         header = f"{self.comment_prefix} {NOTEBOOK_HEADER}"
         if not lines[0].startswith(header):
             raise ValueError("Not a Databricks notebook source!")
 
-        def make_cell(cell_lines: list[str]):
+        def make_cell(cell_lines: list[str], start: int):
             # trim leading blank lines
-            while len(cell_lines) > 0 and len(cell_lines[0]) == 0:
-                cell_lines.pop(0)
+            new_cell_lines = cell_lines
+            while len(new_cell_lines) > 0 and len(new_cell_lines[0]) == 0:
+                new_cell_lines.pop(0)
+                start += 1
+
             # trim trailing blank lines
-            while len(cell_lines) > 0 and len(cell_lines[-1]) == 0:
-                cell_lines.pop(-1)
+            while len(new_cell_lines) > 0 and len(new_cell_lines[-1]) == 0:
+                new_cell_lines.pop(-1)
             cell_language = self.read_cell_language(cell_lines)
             if cell_language is None:
                 cell_language = self
             else:
-                self._remove_magic_wrapper(cell_lines, cell_language)
-            cell_source = '\n'.join(cell_lines)
-            return cell_language.new_cell(cell_source)
+                before = len(cell_lines)
+                new_cell_lines = self._remove_magic_wrapper(new_cell_lines, cell_language)
+                start += before - len(cell_lines)
```

Comment:
```suggestion
                start += before - len(new_cell_lines)
```
Otherwise this is noop (start += 0)

Reply:
YEs - just a hangover from a refactor. FIxed now.